packages/components/inputs/select-utils/src/create-select-styles.ts

```diff
@@ -94,13 +94,17 @@ type TState = {
 
 type TDesignTokenName = keyof typeof designTokens;
 
+const upperFirst = (str: string) => str.charAt(0).toUpperCase() + str.slice(1);
+
 const getHorizontalConstraintValue = (
   value?: TProps['minMenuWidth'] | TProps['maxMenuWidth']
 ) => {
-  return (
-    designTokens[`constraint${value}` as TDesignTokenName] ||
-    designTokens.constraint3
-  );
+  const designTokenSuffix =
+    typeof value === 'string' ? upperFirst(value) : value;
```

Comment:
If the `scale` value is provided, we need a little transformation as its related token name is `constraintScale`.